src/ir/auth_logic/ast_construction_test.cc

```diff
@@ -73,4 +84,62 @@ INSTANTIATE_TEST_SUITE_P(
              datalog::Predicate("mayB", {R"("NotifierB")"},
                                 datalog::Sign::kNegated)})));
 
+TEST_P(SaysAssertionAstConstructionTest, SimpleTestWithSaysAssertionProgram) {
+  const auto& [input_text, principal_name, fact_assertion_name,
+               cond_assertion_predicate, rvalue] = GetParam();
+  Program prog_out = ParseProgram(input_text);
+  ASSERT_EQ(prog_out.says_assertions().size(), 1);
+  const SaysAssertion& says_assertion = prog_out.says_assertions().front();
+  EXPECT_EQ(says_assertion.principal().name(), principal_name);
+  EXPECT_EQ(std::get<datalog::Predicate>(
+                std::get<BaseFact>(
+                    std::get<Fact>(says_assertion.assertions().at(0).GetValue())
+                        .GetValue())
+                    .GetValue())
+                .name(),
+            fact_assertion_name);
```

Comment:
Might be worth adding a comparator for `Predicate`? It is hard to understand what is being verified here.

Reply:
Changed test for assertion to make it simpler for verifying. PTAL